docs/guides/pagination/connections.md

```diff
@@ -0,0 +1,629 @@
+      description="When paginating backwards, are there more items?"
+    )
+
+    start_cursor: Optional[str] = strawberry.field(
+      description="When paginating backwards, the cursor to continue."
+    )
+
+    end_cursor: Optional[str] = strawberry.field(
+      description="When paginating forwards, the cursor to continue."
+    )
+
+
+@strawberry.type
+class Edge(Generic[GenericType]):
+    node: GenericType = strawberry.field(
+      description="The item at the end of the edge."
+    )
+
+    cursor: str = strawberry.field(
+      description="A cursor for use in pagination."
+    )
+
+```
+
+EdgeTypes must have atleast two fields - `cursor` and `node`.
+Each edge has it's own cursor and item (represented by the `node` field).
+
+Now that we have the types needed to implement pagination using Relay Connections, let
+us use them to paginate a list of users. For simplicity's sake, let our dataset be a list of dictionaries.
+
+```py line=7-32
+# example.py
+
+from typing import Generic, TypeVar
+
+import strawberry
+
+user_data = [
+  {
+    "id": 1,
+    "name": "Norman Osborn",
+    "occupation": "Founder, Oscorp Industries",
+    "age": 42
+  },
+  {
+    "id": 2,
+    "name": "Peter Parker",
+    "occupation": "Freelance Photographer, The Daily Bugle",
+    "age": 20
+  },
+  {
+    "id": 3,
+    "name": "Harold Osborn",
+    "occupation": "President, Oscorp Industries",
+    "age": 19
+  },
+  {
+    "id": 4,
+    "name": "Eddie Brock",
+    "occupation": "Journalist, The Eddie Brock Report",
+    "age": 20
+  }
+]
+
+
+GenericType = TypeVar("GenericType")
+
+
+@strawberry.type
+class Connection(Generic[GenericType]):
+    page_info: "PageInfo" = strawberry.field(
+      description="Information to aid in pagination."
+    )
+
+    edges: list["Edge[GenericType]"] = strawberry.field(
+      description="A list of edges in this connection."
+    )
+
+
+@strawberry.type
+class PageInfo:
+    has_next_page: bool = strawberry.field(
+      description="When paginating forwards, are there more items?"
+    )
+
+    has_previous_page: bool = strawberry.field(
+      description="When paginating backwards, are there more items?"
+    )
+
+    start_cursor: Optional[str] = strawberry.field(
+      description="When paginating backwards, the cursor to continue."
+    )
+
+    end_cursor: Optional[str] = strawberry.field(
+      description="When paginating forwards, the cursor to continue."
+    )
+
+
+@strawberry.type
+class Edge(Generic[GenericType]):
+    node: GenericType = strawberry.field(
+      description="The item at the end of the edge."
+    )
+
+    cursor: str = strawberry.field(
+      description="A cursor for use in pagination."
+    )
+
+```
+
+Now is a good time to think of what we could use as a cursor for our dataset. Our cursor needs to be an opaque value,
+which doesn't usually change over time. It makes more sense to use the IDs of the users as our cursor, as it fits both criteria.
```

Comment:
The IDs of the user does not fit the criteria of being opaque.

Reply:
Yes, I'll do that!

Reply:
Done! Thanks for the review!